### ISSUE TITLE
Ids are duplicated across bonds

### ISSUE THREAD

**Issue by plainsane** [aarons22/homebridge-bond#52]
**Describe the bug**
I have 2 bonds and the device Id is duplicated across both my bonds.  

**Expected behavior**
I have 5 fans, HomeKit only shows 4.  

**Screenshots**

Nov 13 00:01:12 zwave homebridge[2186]: [2019-11-13 00:01:12] [Bond] Adding accessory Master Bedroom CF
Nov 13 00:01:12 zwave homebridge[2186]: [2019-11-13 00:01:12] [Bond] Adding accessory Master Bedroom CF
Nov 13 00:01:12 zwave homebridge[2186]: [2019-11-13 00:01:12] [Bond] 00000018 has already been added.
Nov 13 00:01:13 zwave homebridge[2186]: [2019-11-13 00:01:13] [Bond] Adding accessory Office CF
Nov 13 00:01:13 zwave homebridge[2186]: [2019-11-13 00:01:13] [Bond] Adding accessory Media Room CF
Nov 13

**Information (please complete the following information):**
 - Plugin version 2.5.0
 - Bond firmware version 2.9.5
 - iOS version 13.2

**Additional context**
I had sent a pull request to the previous maintainer to update the id of each device to the device attribute that was returned via the api.  This value was a uuid and allowed my kits to work, not sure if they still have that or not.  I’ll dig on it this weekend.
![B1A81F7E-4926-4087-95F3-9EB492ECEB96](https://user-images.githubusercontent.com/755667/68735651-624dc880-05ac-11ea-8edb-ee1901e8b8d5.png)

**Comment by jdlambert** [comment]
Obviously a bug, but one way you could get resolution right now: Re-record the device. The ID on the V2 platform is a random 32-bit hex number, so there's basically no risk of clashing IDs. On the V2 platform it just incremented the ID for every resource created.

**Comment by plainsane** [comment]
ill readd the device this weekend.

**Comment by plainsane** [comment]
Re-aded it, plug-in does not like missing defies subsequent boot, just as a. Heads up.  I had to manually prune my cached accessories

**Comment by chrismerck** [comment]
To be clear, this odd behavior was a "bug" in the Bond firmware (fixed in `v2.10.9`), not in homebridge-bond. However, no negative consequence should have resulted from this. It just violates expectations to see the same `token` in two bonds of the same user.

**Comment by aarons22** [comment]
@chrismerck does this mean re-adding devices will definitely fix the issue?

**Comment by jdlambert** [comment]
@chrismerck this is an issue of a device ID clash over two devices on two Bonds. rather than the duplicated token issue (which we did indeed patch in v2.10.9). This was due to 1) @plainsane using two Bonds on the V1 platform 2) Two devices both being the 18th "thing" created on the platform 3) `homebridge-bond` not playing well with two devices with the same ID

@aarons22 In effect, yes. Technically, the odds of a clash are 1 / (2 ^ 32)... low enough to be negligible at this scale.

Note that there will be many devices like @plainsane's that are "migrated".  That is, they had their IDs created on an earlier platform. They are unique on a per-Bond basis, but may not be unique per User. One way to avoid this clash —  this is how we handle it — is to use the Bond ID and the Device ID together to uniquely identify the device.
